packages/dappmanager/src/modules/docker/utils.ts

```diff
@@ -68,3 +71,55 @@ export function getDockerTimeoutMax(
   }
   return timeout;
 }
+
+/**
+ * Returns the ports value returned by the docker api for a specific container.
+ * Example of ports: Dockerode.Ports[]
+ *   - "IP": "0.0.0.0" => for ipv4
+ *   - "IP": "::" => for ipv6
+ * ```
+ * "Ports": [
+      {
+        "PrivatePort": 80,
+        "Type": "tcp"
+      },
+      {
+        "IP": "0.0.0.0",
+        "PrivatePort": 8000,
+        "PublicPort": 2345,
+        "Type": "tcp"
+      },
+      {
+        "IP": "::",
+        "PrivatePort": 8000,
+        "PublicPort": 2345,
+        "Type": "tcp"
+      }
+    ]
+ * ```
+ */
+export function ensureUniquePortsFromDockerApi(
+  ports: Dockerode.Port[],
+  defaultPorts: PortMapping[] | undefined
+): PortMapping[] {
+  const portsFromApi = ports
+    .filter(port => port.IP && port.IP.trim() !== "::")
+    .map(
+      ({ PrivatePort, PublicPort, Type }): PortMapping =>
+        // "PublicPort" will be undefined / null / 0 if the port is not mapped
+        ({
+          ...(PublicPort ? { host: PublicPort } : {}),
+          container: PrivatePort,
+          protocol: Type === "udp" ? PortProtocol.UDP : PortProtocol.TCP
+        })
+    )
+    .map(
+      (port): PortMapping => ({
+        ...port,
+        deletable: isPortMappingDeletable(port, defaultPorts)
+      })
+    );
+
+  // deduplicate ports
+  return [...new Set(portsFromApi)];
```

Comment:
This does not de-duplicate ports, since different unique objects can represent the same pair of container port and protocol. You must de-duplicate by the key (container port + protocol) as defined in the UI

Reply:
fixed!